bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairBaseThingHandler.java

```diff
@@ -12,24 +12,94 @@
  */
 package org.openhab.binding.pentair.internal.handler;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.binding.pentair.internal.PentairPacket;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SmartHomeUnits;
+import org.openhab.core.thing.Bridge;
 import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
 import org.openhab.core.thing.binding.BaseThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Abstract class for all Pentair Things.
  *
  * @author Jeff James - Initial contribution
  *
  */
+@NonNullByDefault
 public abstract class PentairBaseThingHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(PentairBaseThingHandler.class);
+
     /** ID of Thing on Pentair bus */
     protected int id;
+    protected boolean waitStatusForOnline = false;
 
     public PentairBaseThingHandler(Thing thing) {
         super(thing);
     }
 
+    public void getConfiguration() {
+        id = 0;
+    }
```

Comment:
This method is poorly named, it doesn't retrieve anything and it doesn't look like it relates to the configuration either.